### PR TITLE
CMake: Add CMake support

### PR DESCRIPTION
**Pull request by rajkan01**
- Added CMakeLists.txt file to blockdevice example so that it can build via CMake
- Update mbed-os.lib to latest master to pick CMake changes
- Add CMake test and update Mbed CLI build test in `.travis.yml`

@0xc0170 @hugueskamba

### CODE REVIEW

**Review by hugueskamba — DISMISSED** [review]
LGTM but what determines which examples get to have travis CI? Why wasn't the Mbed OS reference in the lib file not updated? Also shouldn't the commit (and PR description) say "blockdevice example" instead of "filesystem example"?

**Comment by hugueskamba** [comment]
Shouldn't `.travis.yml` be updated to include CMake tests?

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
+language: python sh
```
Comment (hugueskamba) [inline .travis.yml:17]: We need to add the copyright notice here.
Thanks

**Review by 0xc0170 — APPROVED** [review]

**Comment by rajkan01** [comment]
@evedon This PR is approved from CMake team, could you review this PR

**Pull request merged**